### PR TITLE
Individual question answer component

### PR DESCRIPTION
**Pull request by nicholaspix**
- created test for dateParse function 
- created answer list component and individual answer component
- created helpful links for both questions and answers that update using api calls
- created a report link for answers that updates using api calls
- implemented the answer list so that only two answers show initially and gave an ability to expand and collapse additional answers 
- created a sort function that sorts the answer list by helpfulness and if the seller answered

### CODE REVIEW

File: client/src/components/questions/Answer.jsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  var secondSpaceIndex = dateStr.indexOf(' ', 5);
+
+  dateStr = dateStr.slice(0, secondSpaceIndex) + ',' + dateStr.slice(secondSpaceIndex);
+
+  return dateStr;
+};
+
+const Answer = ({ answerData }) => {
+
+  const [reportText, setReportText] = useState(<u>Report</u>);
```
Comment (Nkozlare) [inline client/src/components/questions/Answer.jsx:28]: I didn't realize you could put HTML in the state. That is great. no need to comment on this, you can just close this conversation out.

File: client/src/components/questions/Answer.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        headers: {Authorization: API_KEY}
+      })
+      .then((response) => {
+        setCheckHelpfulClick(true);
+      })
+      .catch((err) => {
+        //console.log(err);
+      })
+    }
+  }
```
Comment (Nkozlare) [inline client/src/components/questions/Answer.jsx:57]: Is the if statement on line 41 just making sure that the function won't be able to send more than one helpful put request?

File: client/src/components/questions/Question.jsx [hunk 3]
```diff
@@ -1,60 +1,84 @@
...
 
-  // turns the answers object into an array for easy iteration
+// turns the answers object into an array and sorts it based on helpfulness and if the seller answered.
+var sortAnswers = (answersObject) => {
   var answerArray = [];
-  for (var key in questionData.answers) {
-    answerArray.push(questionData.answers[key]);
+  for (var key in answersObject) {
+    answerArray.push(answersObject[key]);
+  }
```
Comment (Nkozlare) [inline client/src/components/questions/Question.jsx:28]: If I am reading this correctly, you can use Object.values to automatically create an array of all of the values in an object. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values

File: package.json [hunk 4]
```diff
@@ -26,16 +22,21 @@
...
     "@babel/preset-env": "^7.18.10",
     "@babel/preset-react": "^7.18.6",
     "babel-loader": "^8.2.5",
     "babel-plugin-styled-components": "^2.0.7",
+    "chai": "^4.3.6",
     "css-loader": "^6.7.1",
     "eslint-plugin-react": "^7.30.1",
+    "jest": "^29.0.1",
+    "mocha": "^10.0.0",
+    "nodemon": "^2.0.19",
```
Comment (Nkozlare) [inline package.json:39]: Thanks for moving these over.

**Review by Nkozlare — APPROVED** [review]
Things look good. It looks like your understanding of hooks is pretty solid at this point. I guess in the future, for the sake of reviewing, could you make some comments at the top of each function to explain what it is supposed to do?

**Comment by nicholaspix** [comment]
> Things look good. It looks like your understanding of hooks is pretty solid at this point. I guess in the future, for the sake of reviewing, could you make some comments at the top of each function to explain what it is supposed to do?

Yeah definitely will do going forward.

File: client/src/components/questions/Answer.jsx (hunk 2 above)
Comment (nicholaspix) [reply client/src/components/questions/Answer.jsx:57]: yes, precisely.

File: client/src/components/questions/Question.jsx (hunk 3 above)
Comment (nicholaspix) [reply client/src/components/questions/Question.jsx:28]: Good point. Definitely should add that to my repertoire.

**Pull request merged**